planning/behavior_path_planner/include/behavior_path_planner/scene_module/avoidance/avoidance_utils.hpp

```diff
@@ -73,6 +81,23 @@ std::vector<std::string> getUuidStr(const ObjectDataArray & objs);
 
 Polygon2d createEnvelopePolygon(
   const ObjectData & object_data, const Pose & closest_pose, const double envelope_buffer);
+
+std::vector<Point> getEdgePoints(const Polygon2d & object_polygon, const double threshold);
+
+std::vector<PolygonPoint> getEdgePoints(
+  const std::vector<Point> & bound, const std::vector<Point> & edge_points,
+  const double lat_dist_to_path, size_t & start_segment_idx, size_t & end_segment_idx);
+
+std::vector<PolygonPoint> sortPolygonPoints(const std::vector<PolygonPoint> & points);
+
+std::vector<Point> updateBoundary(
+  const std::vector<Point> & original_bound, const std::vector<PolygonPoint> & points,
+  const size_t start_segment_idx, const size_t end_segment_idx);
```

Comment:
It is better not to use vector as a return value because copying may occur.
https://qiita.com/ysuzuki19/items/df872d91c9c89cc31aee#%E9%96%A2%E6%95%B0%E3%81%AE%E6%88%BB%E3%82%8A%E5%80%A4%E3%81%AB%E3%81%99%E3%82%8B

Reply:
done